app/krs.js

```diff
@@ -47,37 +47,72 @@ const notifyEndpoint = co(function *(key, state) {
   }
 });
 
+const sendDatabaseLowWarning = co(function *(availableKeys, type) {
+  yield utils.sendMailQ(
+    process.config.adminemail,
+    'URGENT: Please replenish the master key database',
+    'databaselow',
+    { availableKeys, type });
+});
+
 /**
  * Selects a random un-assigned master key and sets the coin and customerId fields,
  * returning the key
  * @param coin: coin ticker (btc,eth,etc.)
  * @param customerId: customer ID from the platform
  */
-const provisionMasterKey = co(function *(coin, customerId) {
-  const key = yield MasterKey.findOne({ coin: null, customerId: null });
+const provisionMasterXpub = co(function *(coin, customerId) {
```

Comment:
It might be nicer to keep one func called provisionMasterKey and just look coin up in the config to get the type, and use the type as an arg in places, see comment below as well

Reply:
True, the provisionStellarKey func used to be a lot larger and monolithic until I factored some common logic back to the main func. I didn't notice these two could be merged by the time I was done.